Implement methods to TaxInvoiceModel for json

BeedleEiEi opened the pull request.

Commit:
Add methods for json in TaxInvoiceModel

//Other can be implemented in these methods

Commit:
Add TaxInvoiceModelTests

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Implement-Methods-to-TaxInvoiceModel

champillon commented on src/main/java/th/ac/kmitl/it/soa/group7/models/TaxInvoiceModel.java:
```diff
@@ -1,16 +1,48 @@
 
-	public String getDocumentTypeCode() {// Get typeCode
-		return documentTypeCode;
+	public void setItem(BuyerInformation buyerInformation, SellerInformation sellerInformation) {
```
ทำไม item ต้องใช้ buyer กับ seller information หล่ะ

champillon reviewed (commented):
ชี้แจงมาหน่อย

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Implement-Methods-to-TaxInvoiceModel

Commit:
Change method name

Change name from setIIem to setBuyerAndSellerInformation

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Implement-Methods-to-TaxInvoiceModel

Commit:
Edit test

BeedleEiEi commented:
แก้ให้แแล้วครับ @champillon

champillon reviewed (commented):
เอา PR ใหญ่กว่านี้หน่อยได้มั้ย จะได้เห็น combination ของ data มากกว่านี้

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Implement-Methods-to-TaxInvoiceModel

Commit:
Implement lombok Builder for Json

But it's need to be implemented by another model to be complete json
requirement.

Commit:
Edit tests for json data

This is json format test from the model, But not complete 100% due to
AddressInformation has not been implemented in this pr

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into Implement-Methods-to-TaxInvoiceModel

BeedleEiEi commented:
พี่ @champillon ผมเพิ่มละครับแต่จะมีบางส่วนที่ไม่ได้ถูก implement เพราะยังไม่ได้สร้าง หรือ ยังไม่ผ่าน pr ครับ แล้วก็ แก้ test เพิ่มให้ครับ

champillon commented:
1 คะแนน
